src/iam/certhandler/certhandler.cpp

```diff
@@ -155,28 +153,64 @@ Error CertHandler::GetCertificate(
     return ErrorEnum::eNone;
 }
 
-Error CertHandler::CreateSelfSignedCert(const String& certType, const String& password)
+Error CertHandler::SubscribeCertReceiver(const String& certType, CertReceiverItf& certReceiver)
 {
     LockGuard lock(mMutex);
 
-    LOG_DBG() << "Create self signed cert: type = " << certType;
+    LOG_DBG() << "Subscribe certificate receiver: type = " << certType;
 
     auto* module = FindModule(certType);
     if (module == nullptr) {
         return AOS_ERROR_WRAP(ErrorEnum::eNotFound);
     }
 
-    auto err = module->CreateSelfSignedCert(password);
+    CertInfo certInfo;
+
+    auto err = module->GetCertificate(Array<uint8_t>(), Array<uint8_t>(), certInfo);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    err = mCertReceiverSubscriptions.EmplaceBack(certType, certInfo, &certReceiver);
     if (!err.IsNone()) {
         return AOS_ERROR_WRAP(err);
     }
 
     return ErrorEnum::eNone;
 }
 
-CertHandler::~CertHandler()
+Error CertHandler::UnsubscribeCertReceiver(CertReceiverItf& certReceiver)
 {
-    LOG_DBG() << "Close certificate handler";
+    LockGuard lock(mMutex);
+
+    Error                     err          = ErrorEnum::eNone;
+    CertReceiverSubscription* subscription = nullptr;
+
+    Tie(subscription, err)
+        = mCertReceiverSubscriptions.Remove([&certReceiver](const CertReceiverSubscription& subscription) {
+              return subscription.mReceiver == &certReceiver;
+          });
+
+    return err;
+}
+
+Error CertHandler::CreateSelfSignedCert(const String& certType, const String& password)
+{
+    LockGuard lock(mMutex);
+
+    LOG_DBG() << "Create self signed cert: type = " << certType;
```

Comment:
Remove spaces, so
`Create self signed cert: type='

Reply:
removed in a separate commit for the file